src/fcitx.cpp

```diff
@@ -78,24 +89,90 @@ void start_fcitx() {
     });
 }
 
+void Fcitx::exec() {
+    dispatcher_->attach(&instance_->eventLoop());
+    instance_->exec();
+}
+
+void Fcitx::exit() { instance_->exit(); }
+
+void Fcitx::schedule(std::function<void()> func) {
+    dispatcher_->schedule(func);
+}
+
+fcitx::AddonManager &Fcitx::addonMgr() { return instance_->addonManager(); }
+
+fcitx::AddonInstance *Fcitx::addon(const std::string &name) {
+    return addonMgr().addon(name);
+}
+
+fcitx::MacosFrontend *Fcitx::macosfrontend() {
+    return dynamic_cast<fcitx::MacosFrontend *>(addon("macosfrontend"));
```

Comment:
`macosfrontend` is frequently used, but every call to `addon` queries a map.
https://github.com/fcitx/fcitx5/blob/b56322a85dbc06d74dc855770971d642f69b662e/src/lib/fcitx/addonmanager.cpp#L67
The function could be kept, but the result of `addon("macosfrontend")` should be cached.